provider/pkg/provider/provider.go

```diff
@@ -291,10 +292,30 @@ func (p *googleCloudProvider) Create(ctx context.Context, req *rpc.CreateRequest
 		}
 	default:
 		path := res.CreatePath
+		querySep := "?"
+		if strings.Contains(path, "?") {
+			querySep = "&"
+		}
 		for _, param := range res.CreateParams {
-			key := resource.PropertyKey(param)
+			sdkName := param.Name
+			if param.SdkName != "" {
+				sdkName = param.SdkName
+			}
+			key := resource.PropertyKey(sdkName)
+			if !inputs[key].HasValue() {
+				continue
+			}
+
 			value := inputs[key].StringValue()
-			path = strings.Replace(path, fmt.Sprintf("{%s}", param), value, 1)
+			switch param.Location {
+			case "path":
+				path = strings.Replace(path, fmt.Sprintf("{%s}", param.Name), url.PathEscape(value), 1)
+			case "query":
+				path = fmt.Sprintf("%s%s%s=%s", path, querySep, key, url.QueryEscape(value))
```

Comment:
Any reason not to use something like [url.Url](https://golang.org/pkg/net/url/#URL) to construct the URL?

Reply:
Not really. I wasn't sure how to combine it with replace operations on the path above.

Reply:
Good point! I refactored the code to use `url` to build the query.